packages/components/src/color-picker/alpha.js

```diff
@@ -34,34 +34,58 @@ import { noop } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 import { TAB } from '@wordpress/keycodes';
-import { pure } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { calculateAlphaChange } from './utils';
 import KeyboardShortcuts from '../keyboard-shortcuts';
 
-export class Alpha extends Component {
-	constructor() {
-		super( ...arguments );
+export default function Alpha( { rgb, hsl, onChange = noop } ) {
+	const [ mouseDown, setMouseDown ] = useState( false );
+	const container = useRef();
 
-		this.container = createRef();
-		this.increase = this.increase.bind( this );
-		this.decrease = this.decrease.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.handleMouseUp = this.handleMouseUp.bind( this );
+	const rgbString = `${ rgb.r },${ rgb.g },${ rgb.b }`;
+	const gradient = {
+		background: `linear-gradient(to right, rgba(${ rgbString }, 0) 0%, rgba(${ rgbString }, 1) 100%)`,
+	};
+	const pointerLocation = { left: `${ rgb.a * 100 }%` };
+
+	function handleChange( e ) {
+		const change = calculateAlphaChange( e, { hsl }, container.current );
+		if ( change ) {
+			onChange( change, e );
+		}
+	}
+
+	function handleMouseUp() {
+		setMouseDown( false );
 	}
 
-	componentWillUnmount() {
-		this.unbindEventListeners();
+	function preventKeyEvents( event ) {
+		if ( event.keyCode === TAB ) {
+			return;
+		}
+		event.preventDefault();
 	}
 
-	increase( amount = 0.01 ) {
-		const { hsl, onChange = noop } = this.props;
+	useEffect( () => {
+		const { ownerDocument } = container.current;
+
+		if ( mouseDown ) {
+			ownerDocument.addEventListener( 'mousemove', handleChange );
+			ownerDocument.addEventListener( 'mouseup', handleMouseUp );
+		}
+
+		return function cleanup() {
```

Comment:
We should only clean up if `mouseDown` is true. I would suggest to return early if `mouseDown` is false.

```js
if ( mouseDown ) {
  return;
}

ownerDocument.addEventListener( 'mousemove', handleChange );

return () => {
  ownerDocument.removeEventListener( 'mousemove', handleChange );
}
```

In this case we're returning the function, so it's ok for it to be anonymous. The clean up function is anonymous everywhere in the codebase.

Reply:
So, regarding the named functions, I only do it if you'd otherwise assign it to a constant. E.g. `const onChange = () => ...`.

Reply:
I agree it's much better like this. i made the changes in daddou-ma/gutenberg@2bbe328